File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletCO2V2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletCO2V2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your CO2 Bricklet 2.0

$ipcon = new IPConnection(); // Create IP connection
$co2 = new BrickletCO2V2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current all values
$all_values = $co2->getAllValues();

echo "CO2 Concentration: " . $all_values['co2_concentration'] . " ppm\n";
echo "Temperature: " . $all_values['temperature']/100.0 . " °C\n";
echo "Humidity: " . $all_values['humidity']/100.0 . " %RH\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{co2_v2_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your CO2 Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let co2 = Co2V2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current all values.
    let all_values = co2.get_all_values().recv()?;

    println!("CO2 Concentration: {} ppm", all_values.co2_concentration);
    println!("Temperature: {} °C", all_values.temperature as f32 / 100.0);
    println!("Humidity: {} %RH", all_values.humidity as f32 / 100.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
